### PR TITLE
Edit Site: Avoid dirtying un-customized templates on first load.

### PR DESCRIPTION
**Pull request by epiqueras**
## Description

This PR stops uncustomized templates from becoming dirty just by being loaded in the site editor. It does this by ensuring template parts don't persist a post ID unless they've been edited, and by deferring the site editor's initial status and title template edits to the entities saved states callback, which is now called right before saving.

## How has this been tested?

It was verified that loading the site editor on a template without saved customizations, that contains template parts, does not make it dirty.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Review by noahtallen — CHANGES_REQUESTED** [review]
I don't think this is quite working for me:
1. Template Parts are still dirty (I know this isn't the goal of the PR, but would still be nice to solve)
2. The first edit to a template part dirties the template as well. (probably because of the template part ID being set?)

It seems like the solution here is just deferring the problem, so I wonder if there is a more robust way to solve it.

I tested this by cleaning my environment and re-enabling the FSE experiment & templates before testing

File: packages/block-library/src/template-part/edit/index.js [hunk 1]
```diff
@@ -22,17 +24,27 @@ export default function TemplatePartEdit( {
 	// Resolve the post ID if not set, and load its post.
 	const postId = useTemplatePartPost( _postId, slug, theme );
 
-	// Set the post ID, once found, so that edits persist.
+	// Set the post ID, once found, so that edits persist,
+	// but wait until the third inner blocks change,
+	// because the first 2 are just the template part
```
[the remark this answers is not in the document]
Comment (epiqueras) [reply packages/block-library/src/template-part/edit/index.js:29]: 1. Empty block list.
2. Loaded block list.

**Comment by epiqueras** [comment]
>Template Parts are still dirty (I know this isn't the goal of the PR, but would still be nice to solve)

When?

>The first edit to a template part dirties the template as well. (probably because of the template part ID being set?)

Yes, this is still a clear improvement over `master`. Do you think we can merge this as a first step?

**Comment by noahtallen** [comment]
> When?

On loading the editor with non-customized template parts (same situation as when the template would be dirtied).

Obviously none of this is an issue once we are loading from CPTs.

> this is still a clear improvement over `master`

I think I'm onboard with this if we fix the same problem with template parts. Otherwise, it's not really noticeable that anything has been improved, since a dirty state is still visible when entering the editor for the first time... So on load, I get:

Dirtied: template part

Then, I modify the template part and I get:

Dirtied: template and template part

This is confusing behavior to me, and I feel like the template & template part being dirty the whole time makes more sense, in a way.

The other thing is that I assume if we fix the problem in a more robust way, then we wouldn't need this workaround

**Comment by epiqueras** [comment]
>On loading the editor with non-customized template parts (same situation as when the template would be dirtied).
>I think I'm onboard with this if we fix the same problem with template parts.

Fixed :)

**Review by noahtallen — APPROVED** [review]
Nice! works for me. Now the editor shows no dirty changes until you actually modify something, so it at least looks like expected behavior now.

**Comment by youknowriad** [comment]
This is a really cool one. It fixes one of the most important bugs in the current flows.

**Pull request merged**